### PR TITLE
dialects: Tobias stencil memref.subview

### PR DESCRIPTION
**Pull request by tobiasgrosser**
Rebased https://github.com/xdslproject/xdsl/pull/660 on main with @georgebisbas recent test cases from https://github.com/xdslproject/xdsl/pull/662.

### CODE REVIEW

**Review by georgebisbas — APPROVED** [review]

File: tests/filecheck/mlir-conversion/with-mlir/dialects/stencil/hdiff.mlir
Comment (georgebisbas) [inline tests/filecheck/mlir-conversion/with-mlir/dialects/stencil/hdiff.mlir:1]: I wonder whether we would not like to drop the previous tests.
As in testing workflows, with and without subviews.

Comment (georgebisbas) [reply tests/filecheck/mlir-conversion/with-mlir/dialects/stencil/hdiff.mlir:1]: This is what I am wondering, should it be te only option?

**Review by georgebisbas — COMMENTED** [review]
@PapyChacal did we text the correctness of this PR with Devito?

**Comment by PapyChacal** [comment]
@georgebisbas Yes!

**Review by georgebisbas — APPROVED** [review]

File: tests/filecheck/mlir-conversion/with-mlir/dialects/stencil/hdiff.mlir
Comment (PapyChacal) [reply tests/filecheck/mlir-conversion/with-mlir/dialects/stencil/hdiff.mlir:1]: Oh, sorry I missed your question.
I don't think it make much sense to make it an option, now that we can use subview. subview is just a higher-level representation of this offset we did, that will be lowered to the same code - if not handled even better in some way by MLIR! .
More importantly, it allows to use other higher-level facilities on this subview, such as memref.copy, gpu.memcpy, ...

**Pull request merged**